### PR TITLE
Feature: Add custom sidebar hook and component

### PR DESCRIPTION
**Pull request by mayan-000**
## Description
This PR adds a new generic Sidebar component and hook to the CLI dashboard for managing multi-level sidebar. Reusing the extension's sidebar was getting hacky and specific to the extension's environment.

## Relevant Technical Choices
- The type `SidebarItems` is created as a disjoint tree-based structure that takes values and children, keeping in mind for easy lookup of data and better visual perspective from the data provided.
- The hook `useSidebar` takes in data, with a specified type, for sidebar items that need to be manipulated and methods to handle state changes.
- The state and callbacks provided by `useSidebar` are passed on to the `Sidebar` component as props, keeping in mind that the developer can wrap/modify the callbacks up to their needs before passing them as props and listening to state changes.

## Additional Information:
This sidebar is currently present in cli-dashboard and not in the design-system as the sidebar currently does not have the capability for keyboard navigation.
When keyboard navigation gets implemented in the future, we can reuse it in the extension too.

## Screenshot/Screencast

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/58820001/2874aa8f-58ff-4f2e-aa9b-a1d55f96b731

---

Partially addresses #206

### CODE REVIEW

File: packages/design-system/src/components/sidebar/useSidebar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+interface useSidebarProps {
+  data: SidebarItem[];
+}
+
+const useSidebar = ({ data }: useSidebarProps): SidebarOutput => {
+  const [selectedItemKey, setSelectedItemKey] = useState<string | null>(null);
+  const [activePanel, setActivePanel] = useState<React.ReactNode>();
+  const [sidebarItems, setSidebarItems] = useState<SidebarItem[]>([]);
```
Comment (ayushnirwal) [inline packages/design-system/src/components/sidebar/useSidebar/index.tsx:46]: `data` can be passed as an argument to useState hook here

File: packages/design-system/src/components/sidebar/useSidebar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+  useEffect(() => {
+    setSidebarItems(data);
+  }, [data]);
+
+  const matchKey = useCallback((key: string, toMatch: string) => {
```
Comment (ayushnirwal) [inline packages/design-system/src/components/sidebar/useSidebar/index.tsx:52]: This function does not use any state or ref vars please consider moving it out of the component.

Comment (mayan-000) [reply packages/design-system/src/components/sidebar/useSidebar/index.tsx:52]: This change has already been made in PR #266.

File: packages/design-system/src/components/messageBox/index.tsx [hunk 3]
```diff
@@ -24,7 +24,7 @@ interface MessageBoxProps {
 }
 const MessageBox = ({ headerText, bodyText }: MessageBoxProps) => {
   return (
-    <div className="bg-hsl-light dark:hsl-dark p-4 m-3 leading-5 max-w-2xl">
+    <div className="bg-hsl-light dark:hsl-dark py-4 my-3 leading-5 max-w-2xl">
```
[the remark this answers is not in the document]
Comment (mayan-000) [reply packages/design-system/src/components/messageBox/index.tsx]: Thanks, yes this is not related.

Removing it.

File: packages/cli-dashboard/src/components/siteReport/components/layout.tsx [hunk 4]
```diff
@@ -16,77 +16,117 @@
...
+    }
+  }, [selectedItemKey, updateSelectedItemKey]);
 
   return (
     <div className="w-full h-full flex">
       <Resizable
         defaultSize={{ width: '200px', height: '100%' }}
         minWidth={'150px'}
-        maxWidth={'98%'}
+        maxWidth={'60%'}
```
[the remark this answers is not in the document]
Comment (mayan-000) [reply packages/cli-dashboard/src/components/siteReport/components/layout.tsx:117]: The decrease in max-width is required as the Sidebar squeezed the Panel too much when resized by the user.

**Pull request merged**